From 978585b0443db13ea6c7fb3b6e22701b1cb5b3b4 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 19 Dec 2021 20:50:23 +0000
Subject: [PATCH 1/4] gnu: Add Golang module - brotli

* gnu/packages/golang.scm: (go-github-com-andybalholm-brotli): New variable.
---
 gnu/packages/golang.scm | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index c863388475..134bc8f16a 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -8832,3 +8832,33 @@ (define-public go-github-com-arceliar-ironwood
 @url{https://github.com/yggdrasil-network/yggdrasil-go,Yggdrasil}, but it may
 be useful for other network applications.")
     (license license:mpl2.0)))
+
+(define-public go-github-com-andybalholm-brotli
+  (package
+    (name "go-github-com-andybalholm-brotli")
+    ;; NOTE: (Sharlatan-20211218T165057+0000): v1.0.4 requires Go 1.16+ for
+    ;; io.ReadAll
+    ;; https://github.com/andybalholm/brotli/blob/v1.0.4/brotli_test.go#L122
+    ;; https://go.dev/doc/go1.16#ioutil
+    (version "1.0.2")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/andybalholm/brotli")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0l2z836kwdf3qbmajh97bzl476zh88pr209pn5kxhd0gw3jkdrib"))))
+    (build-system go-build-system)
+    (arguments
+     `(#:import-path "github.com/andybalholm/brotli"))
+    (home-page "https://github.com/andybalholm/brotli")
+    (synopsis "Pure Go Brotli encoder and decoder")
+    (description
+     "This package is a brotli compressor and decompressor implemented in Go.  It was
+translated from the reference implementation
+(@url{https://github.com/google/brotli,https://github.com/google/brotli}) with
+the @code{c2go} tool at
+@url{https://github.com/andybalholm/c2go,https://github.com/andybalholm/c2go}.")
+    (license license:expat)))
-- 
2.34.0

